blocks/library/freeform/format-list.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+import classnames from 'classnames';
+import { fromPairs, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Dashicon } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import './format-list.scss';
+
+function naiveCss2Jsx( styleText ) {
+	return fromPairs(
+		styleText.split( ';' ).map(
+			( stylePart ) => {
+				const [ cssKey, cssValue ] = stylePart.split( ':', 2 );
+				const reactKey = cssKey.split( '-' ).map(
+					( part, index ) => (
+						index === 0 ?	part : part.substr( 0, 1 ).toUpperCase() + part.substr( 1 )
+					)
+				).join( '' );
+				return [ reactKey, cssValue ];
+			}
+		)
+	);
+}
+
+class FormatList extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.switchFormat = this.switchFormat.bind( this );
+		this.toggleMenu = this.toggleMenu.bind( this );
+		this.state = {
+			open: false,
+		};
+	}
+
+	handleClickOutside() {
+		if ( ! this.state.open ) {
+			return;
+		}
+		this.setState( { open: false } );
+	}
+
+	toggleMenu() {
+		this.setState( {
+			open: ! this.state.open,
+		} );
+	}
+
+	switchFormat( newValue ) {
+		if ( this.props.onFormatChange ) {
+			this.props.onFormatChange( newValue );
+		}
+		this.setState( { open: false } );
+	}
+
+	render() {
+		const { formats } = this.props;
+		const selectedValue = this.props.value;
+		const noFormat = { text: wp.i18n.__( 'No format' ), value: null };
```

Comment:
Could we use the gutenberg `wp` modules consistently? I mean avoid using the global `wp` and use `import { __ } from 'i18n';`?